app/scripts/directives/overview/mobileClientRow.js

```diff
@@ -22,12 +21,27 @@
     templateUrl: 'views/overview/_mobile-client-row.html'
   });
 
-  function MobileAppRow($scope, $filter, $routeParams, APIService, AuthorizationService, DataService, ListRowUtils, Navigate, ProjectsService) {
+  function MobileAppRow($filter, $routeParams, APIService, AuthorizationService, DataService, ListRowUtils, Navigate, ServiceInstancesService) {
     var row = this;
+    var serviceInstancesVersion = APIService.getPreferredVersion('serviceinstances');
+    var isServiceInstanceReady = $filter('isServiceInstanceReady');
+    var isMobileService = $filter('isMobileService');
     row.installType = '';
 
     _.extend(row, ListRowUtils.ui);
 
+    row.$onInit = function() {
+      row.context = {namespace: _.get(row, 'apiObject.metadata.namespace')};
+      DataService.watch(serviceInstancesVersion, row.context, function (serviceinstances){
+        row.services = _.filter(serviceinstances.by('metadata.name'), function(serviceInstance){
+          return ServiceInstancesService.fetchServiceClassForInstance(serviceInstance)
```

Comment:
This is going to repeatedly request service classes anytime the instance changes. We should only request it one time.